importer/Monument.py

```diff
@@ -262,49 +258,70 @@ def create_wlm_source(self, monuments_all_id):
                 "reference_url": {"prop": prop_reference_url,
                                   "value": self.wlm_url}}
 
+    def exists_with_monument_article(self,
+                                     language,
+                                     article_keyword="monument_article"):
+        """Check whether the data object has anythin in the monument_article field."""
+        if self.has_non_empty_attribute(article_keyword):
+            wd_item = utils.q_from_wikipedia(
+                language, getattr(self, article_keyword))
+            return wd_item
+        else:
+            return None
+
+    def exists_with_wd_item(self, wd_item_keyword="wd_item"):
+        """Check whether the data object has anything in the 'wd_item' field."""
+        if wd_item_keyword in self.raw_data:
+            return self.raw_data[wd_item_keyword]
+        else:
+            return None
+
     def exists_with_prop(self, mapping):
-        if self.existing is None:
-            return
+        """
+        Check whether there already exists an item
+        with the same unique ID as that in the data object.
+        """
         unique_prop = mapping.get_unique_prop()
         base = self.wd_item["statements"]
         if unique_prop in base:
             val_to_check = base[unique_prop][0]['value']
             if val_to_check in self.existing:
                 wd_item = self.existing[val_to_check]
-                print("Wikidata has item with {} = {}. Connecting with item {}.".format(
-                    unique_prop, val_to_check, wd_item))
-                # Check whether it already has self.wd_item
-                # from monument_article.
-                # This of course assumes that self.exist is run earlier.
-                # If it's a different one,
-                wd_item_from_monument_article = self.wd_item["wd-item"]
-                if wd_item != wd_item_from_monument_article:
-                    # different Q-numbers from monuments_all and
-                    # unique property
-                    self.wd_item["upload"] = False
-                else:
-                    pass  # match
-                self.set_wd_item(wd_item)
+                return wd_item
             else:
-                print("There's no item with {} = {} on Wikidata.".format(
-                    unique_prop, val_to_check))
+                return None
+
+    def in_known_items(self, wd_item):
+        """Check whether given item is on the list of items with certain prop"""
+        return wd_item in self.existing.items()
+
+    def find_matching_wikidata(self, mapping):
```

Comment:
Is there a need to have this return the `wd_item` rather than always passing it on to `self.set_wd_item()`?